### DISCUSSION TITLE
Maximum file size cannot be set for `FileUpload` component

### DISCUSSION THREAD

**Discussion by alexmanase** [filamentphp/filament#2711] [Help] [answered]
### Package

filament/filament

### Package Version

v2.12.28

### Laravel Version

v9.16

### Livewire Version

v2.10

### PHP Version

PHP 8.1.6

### Bug description

When I try to upload a larger file I get an error in console: `422 (Unprocessable Content)`. I set the limit for for max file size to 102400 KB, but it does not upload a file with size of 2.4 MB.

![image](https://user-images.githubusercontent.com/10696975/172813314-208771bf-e792-415b-92a6-3c83435ad1e7.png)

For smaller images it works perfectly, but in an ideal world it may be needed for larger images.

### Steps to reproduce

**1. Use the `FileUpload` component**

```php
public static function form(Form $form): Form
{
    // ...
    FileUpload::make('image')
        ->maxSize(102400)
        ->label('Featured Image')
        ->image()
}
```

**2. Upload a large file (2 MB or greater)**

### Relevant log output

```shell
There is only one error in console:

`POST http://cms.test/livewire/upload-file?expires=1654702019&signature=df75b5ad101ec6b07d69162b79e3e21647abf5673d8ed49cf96f8b229e4dcdd8 422 (Unprocessable Content)`
```

**Comment by danharrin** [comment, marked as answer]
Does this help you? https://stackoverflow.com/a/2184541

The console error is from Livewire, not Filament.

**Comment by alexmanase** [comment]
I already tried that and didn't work.

**Comment by danharrin** [comment]
Unfortunately, I don't think this is something wrong with Filament, it is likely your server setup or Livewire

**Reply by alexmanase** [reply]
Is it fine if I create a Laravel project with this issue to make sure that it's not my server setup?

If it is a problem with Livewire, I will open an issue there.

**Reply by danharrin** [reply]
Yes, please create a new Laravel project with a file upload inside a Livewire component 👍

**Reply by alexmanase** [reply]
https://github.com/alexmanase/filament-upload-file-bug-demo

**Reply by danharrin** [reply]
I meant try using a file upload outside the form builder, without Filament, using Livewire

**Reply by alexmanase** [reply]
Got it.

**Reply by alexmanase** [reply]
Yeah, I think it's a Livewire bug. Can you please check on your machine, in order to be sure?

https://github.com/alexmanase/upload-file-bug-demo

**Reply by alexmanase** [reply]
I checked on somebody's computer and it looks like the problem is with my setup. Thank you for your help!

**Comment by DenisaHalmaghi** [comment]
I faced the exact same issue, this are the steps I took to make it work:

1. Publish the Livewire config
`php artisan livewire:publish --config`

2. In `config/livewire.php`, modify the validation rules for temporary file uploads:
```php
    'temporary_file_upload' => [
        ...
        'rules' => 'file|max:102400',
    ],
```

**Reply by MOHAMMED-ABD-RAZAQ** [reply]
thanks , this works for me

**Reply by YousefAlamir** [reply]
Same here, It worked.
**Thanx**

**Reply by cryswerton** [reply]
Thank you! It worked!

**Reply by zuhrrl** [reply]
Thanks it's worked!

**Comment by adevade** [comment]
I ran into this as well. Turns out it was my local nginx config. 

My error was `413 Request Entity Too Large` though.

Setting `client_max_body_size` in the server config fixed it!

```conf
http {
    client_max_body_size 64M;

    // ...
}
```

The default is `1M`, so not a lot.

[Source on Stack Overflow](https://stackoverflow.com/a/23764479)

**Reply by VicChaos32** [reply]
This worked for me after setting php.ini alone didn't. Thanks.

```
php.ini
; Maximum allowed size for uploaded files.
upload_max_filesize = 40M

; Must be greater than or equal to upload_max_filesize
post_max_size = 40M
```

Some suggest setting post_max_size larger than upload_max_size since actual payload can be larger than the file itself.

**Comment by dev7771** [comment]
<img width="604" height="234" alt="Screenshot 2025-08-01 at 17 06 26" src="https://github.com/user-attachments/assets/3e35784e-bcb5-4edf-af76-ba98d91b5d7e" />
vendor/livewire/livewire/src/Features/SupportFileUploads/FileUploadConfiguration.php

Change it

**Comment by hugolb1000** [comment]
Error en esta imagen al subir 
<img width="1281" height="672" alt="IMG-20260917-WA0012" src="https://github.com/user-attachments/assets/0c415735-7c38-43e6-a032-527acfeffae7" />
